### PR TITLE
Core/Unit: Some improved function RemoveAuraFromStack

### PR DESCRIPTION
**Pull request by zgn1988**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  improved function RemoveAuraFromStack

**Target branch(es):** 3.3.5/master

- [x] master

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -3645,7 +3645,7 @@ void Unit::RemoveAurasDueToSpell(uint32 spellId, ObjectGuid casterGUID, uint32 r
     }
 }
 
-void Unit::RemoveAuraFromStack(uint32 spellId, ObjectGuid casterGUID, AuraRemoveMode removeMode)
+void Unit::RemoveAuraFromStack(uint32 spellId, ObjectGuid casterGUID, AuraRemoveMode removeMode, int32 num)
```
Comment (P-Kito) [inline src/server/game/Entities/Unit/Unit.cpp]: Can auras stack more than 255 stacks? If not you could use int8. 
Just a random idea :P

Comment (robinsch) [inline src/server/game/Entities/Unit/Unit.cpp]: uint8

Comment (Riztazz) [inline src/server/game/Entities/Unit/Unit.cpp]: 72963 can stack up to 500, not sure if TC can handle anything beyond 255 tho :P

**Comment by joschiwald** [comment]
in db2 stackamount has type uint16

**Comment by zamalaev** [comment]
@joschiwald need uint8 to uint16? finish fix

**Comment by joschiwald** [comment]
uint16 better than uint8

**Pull request merged**